sections/3-architecture.md

```diff
@@ -366,3 +366,107 @@ primitives (e.g. ints, strings, etc.), so usage of named
 parameters makes the code more resistant to change and less error-prone,
 versus relying on positioning. The style of indentation chosen here makes
 the instantiation look like a `Map` or a JSON object if you want.
+
+### 3.6 SHOULD put type-class instances as close to the call site as possible
+
+Type-classes are a powerful feature close to functional programming mindset. It 
+favours composition and is central to core `Scala` libraries like 
+[`circe`](https://circe.github.io/circe/), [`cats`](https://typelevel.org/cats/) 
+or even [`scalacheck`](https://www.scalacheck.org/).
+
+One may keep wondering where to put the type-class instances. Each type-class
+has a purpose. Some may be more general like purely functional abstractions 
+from `cats`, some more specific like *json* serialization from `circe`. A good 
+practice is to *put them as close to the call site as possible*. For example:
```

Comment:
Worth mentioning default implicit scope and priorities like import to local scope cannot be replaced etc. So default go to companion objects, package objects ... follows natural code structure.